src/wp-admin/includes/image-edit.php

```diff
@@ -873,18 +904,32 @@ function wp_save_image( $post_id ) {
 		return $return;
 	}
 
-	$meta         = wp_get_attachment_metadata( $post_id );
-	$backup_sizes = get_post_meta( $post->ID, '_wp_attachment_backup_sizes', true );
+	$meta           = wp_get_attachment_metadata( $post_id );
+	$backup_sizes   = get_post_meta( $post->ID, '_wp_attachment_backup_sizes', true );
+	$backup_sources = get_post_meta( $post->ID, '_wp_attachment_backup_sources', true );
```

Comment:
We should use `$post_id` instead of `$post->ID`? Is there any difference in post IDs?

Reply:
I think it is fine to use both `$post_id` & `$post->ID` there. Previously it was using `$post->ID` for the `$backup_sizes`.

Reply:
It makes sense, we can use both as per Sergey's comment here: https://core.trac.wordpress.org/ticket/22372#comment:1